### PR TITLE
Upgrade to Rust 1.43.0

### LINKED ISSUE #345: Upgrade to Rust 1.43.0
https://blog.rust-lang.org/2020/04/23/Rust-1.43.0.html

How to: https://github.com/CodeChain-io/foundry/pull/259

### PR DESCRIPTION
**Pull request by HoOngEe**
it resolves #345

### CODE REVIEW

File: core/src/client/test_client.rs [hunk 1]
```diff
@@ -438,7 +438,7 @@ impl ImportBlock for TestBlockChainClient {
         }
         let len = self.numbers.read().len();
         if number == len {
-            mem::replace(&mut *self.last_hash.write(), h);
```
Comment (sgkim126) [inline core/src/client/test_client.rs:441]: How about using `swap`?

Comment (sgkim126) [reply core/src/client/test_client.rs:441]: https://doc.rust-lang.org/std/mem/fn.swap.html

Comment (HoOngEe) [reply core/src/client/test_client.rs:441]: If we use swap here, the following lines are affected from the content swap.
This function returns the value h as `Ok(h)`, `swap` will produce unwanted changes in code.

Comment (sgkim126) [reply core/src/client/test_client.rs:440]: I think it's okay if we call swap on the cloned `h`, but I'm not sure it's more readable. Let's go ahead with this code.

**Pull request merged**